src/thor/costmatrix.cc

```diff
@@ -728,7 +728,7 @@ void CostMatrix::SetSources(GraphReader& graphreader,
       // Set the predecessor edge index to invalid to indicate the origin
       // of the path.
       uint32_t idx = source_edgelabel_[index].size();
-      source_edgelabel_[index].push_back(std::move(edge_label));
```

Comment:
i thought the wisdom was kind of like pass by reference, doesnt make sense for primitives and small PODs but for larger types it still makes sense. now i think edge_lables are PODs but they are a non trivial amount of members iirc. i wonder if there is a performance impact not using copy elision in such tight loops. and those in the rest of the diff. someone should test this before we go with it. im all for the simplification but not at the sake of performance

Reply:
This change is listed under tidy's performance checks but could certainly be disabled. It's possible clang and gcc differs in these optimizations.

I can run some tests to dig into more details.